src/clarity-angular/popover/dropdown/dropdown.ts

```diff
@@ -15,10 +18,25 @@ import {IfOpenService} from "../../utils/conditional/if-open.service";
         // FIXME: remove this as soon as we stop supporting this old <div class="dropdown-menu"> syntax
         "[class.open]": "ifOpenService.open"
     },
-    providers: [IfOpenService]
+    providers: [
+        IfOpenService, {
+            provide: DropdownService,
+            useFactory: clrDropdownProviderFactory,
+            deps: [[new Optional(), new SkipSelf(), DropdownService]]
+        }
```

Comment:
If you do end up keeping a separate file, please move the actual Provider into it, rather than just the factory. The current halfway split doesn't make much sense. I know we did the same in TreeView, but I didn't pick it up during the review.